### ISSUE TITLE
dotty doc fails on ScalaCheck project

### ISSUE THREAD

**Issue by ashawley** [lampepfl/dotty#7326]
I was able to get the ScalaCheck code base to compile with Dotty 0.19.0-RC1, but dotty doc fails to run.  It's choking on the `UserGuide.md` file.  You can see it happening here:

https://github.com/typelevel/scalacheck/pull/565

Here's the failing Travis job:

https://travis-ci.org/typelevel/scalacheck/jobs/589777956

I tried to create a minimal example by adding a `UserGuide.md`, but dotty doc works fine:

https://github.com/ashawley/dotty-example-project/pull/1

I predict there is something strange about the ScalaCheck build configuration in sbt that is causing the `doc` task to fail for dotty.

**Comment by TheElectronWill** [comment]
I'll take a look

**Comment by TheElectronWill** [comment]
Here is what seems to happen: with scalacheck's sbt configuration, Dottydoc runs in the project root directory. It tries to evaluates every markdown file as a [liquid template](https://shopify.github.io/liquid/basics/introduction/), where expressions like `{{ something }}` or `{% something %}` are significant.

At line 906 in `UserGuide.md`, there is this:
```scala
   *  {{{
   *  def canCreateNewSut(newState: State, initSuts: Traversable[State]
   *    runningSuts: Traversable[Sut]
   *  ) = {
   *    initSuts.isEmpty && runningSuts.isEmpty
   *  }
   *  }}}
```
It's a perfectly fine scaladoc, but it contains double-braces that liquid tries (and fails) to evaluate.

Removing the braces makes the error go away, but the `doc` task should probably pass a `-siteroot actualDocTemplatesDir` option to Dottydoc so that it doesn't try to evaluate the user guide.

Btw the error that has been reported is both unclear (`token 57`??) and misleading (wrong line), I'll open a PR to fix this.

**Comment by smarter** [comment]
Can we turn this error into a warning ? That would be the simplest fix for now until we decide how dottydoc should behave (default directory, default way to handle markdown files, etc)

**Comment by ashawley** [comment]
This is not a blocker for ScalaCheck.  I believe we could temporarily disable the doc task for Dotty, so it's fine to leave this issue open and address the defects.

**Comment by ashawley** [comment]
I'm not sure when this was fixed, but I checked 3.0.0-RC2 and it doesn't appear to be a problem.

https://github.com/ashawley/dotty-example-project/pull/2
